pkg/apis/resource/types.go

```diff
@@ -222,6 +222,57 @@ type PodSchedulingStatus struct {
 	// for those other resources.
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PodSchedulingHints objects hold information that is needed to schedule
+// a Pod with ResourceClaims that use "WaitForFirstConsumer" allocation
+// mode.
+//
+// This is an alpha type and requires enabling the DynamicResourceAllocation
+// feature gate.
+type PodSchedulingHints struct {
```

Comment:
"Hint" implies "optional" - is that really the case here?

Reply:
It's more "hint" in the sense of "potentially incomplete information" than "feel free to ignore this".

Some fields are required. Without `spec.selectedNode`, delayed allocation will not start.

See https://github.com/kubernetes/kubernetes/issues/114283 for some other ideas. We can continue brainstorming...

- `PodSchedulingInfo`
- `PodSchedulingData`

In both cases, we probably want to use the same word for plural and singular?

/cc @sftim